addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -652,7 +652,17 @@ odoo.define('website.s_website_form', function (require) {
                 }
 
                 const formData = new FormData(this.$target[0]);
-                const currentValueOfDependency = formData.get(dependencyName);
+                let currentValueOfDependency = formData.get(dependencyName);
+                if (currentValueOfDependency === null) {
+                    // TODO One customer apparently reached that case but not
+                    // sure how this is possible. In any case, it should not
+                    // make the form crash: it's just a problematic *visibility*
+                    // dependency, it should not prevent using the form. For
+                    // now, let's just add an error in the console so we may
+                    // investigate the issue if it ever shows up in a test.
+                    console.error("This field dependency cannot be found");
+                    currentValueOfDependency = '';
+                }
```

Comment:
@qsm-odoo Here is what I understood for this issue:
There are indeed some strange things about the public widget of conditional visibilities. 
Firstly, the `formData` object which contains key-value pairs for the form fields does **not** contain checkboxes or radios which are not checked / selected. So with a form like this:
```html
<form>
        <input type="checkbox" name="My Checkbox"/>
</form>
```
If we do this
```js
const formData = new FormData(document.querySelector('form'));
console.log(formData.get('My Checkbox'));
```
this will print `null`. This is not the case for other input types text with a `value=''` or even which has no value attribute.

Secondly, I didn't know this, but when you have a function with a default value for a certain param like this
```js
function print(value = '') {
	console.log(value)
}
```
and you call it with null like this `print(null)` , the default value won't override and the function will log `null`.

Despite these two strange things which make the code a bit fragile IMO, I don't see how the users could have ended up with the traceback they mention on the ticket because the only calls to `.includes()` in the `compareTo()` function are only called if the `comparator` is `contains` or `!contains` which is `only` possible on inputs with type in `['text', 'email', 'tel', 'url', 'search', 'password']` or on textarea (I checked each of these elements and the formData returned an empty string if they weren't filled in). The other `.includes()` is called for dates (or an unrecognised comparator) and again, `formData` returns an empty string for dates without value.

I also tested the case where we put a field A that has visibility that depends on a field B, if we delete field B, we delete the visibility condition on field A. So that is working as expected.

With all these elements, I think that either the client has done something weird with the form by modifying the HTML himself or an upgrade script has broken the form...

So, in my opinion, the only thing that would seem a good thing to do is to change the value to `''` if `null` is passed as a parameter of `compareTo`. It's not supposed to be a problem but it would make the code more robust.

let me know if I can do anything or if something is not clear :slightly_smiling_face: